de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/server/rpl/imp/RabbitMqListener.java

```diff
@@ -0,0 +1,306 @@
+					+ "\n Processor: " + replicationProcessor
+					, e);
+		}
+	}
+
+	private void run0() throws Exception
+	{
+
+		connectionFactory = new CachingConnectionFactory(host, port);
+		if (userName != null && password != null)
+		{
+			connectionFactory.setUsername(userName);
+			connectionFactory.setPassword(password);
+		}
+		RabbitAdmin admin = new RabbitAdmin(connectionFactory);
+		Queue queue = new Queue(queueName, isDurableQueue);
+		exchangeName = "de.metas.esb.durable.exchange";
+		TopicExchange exchange = new TopicExchange(exchangeName, isDurableQueue, false);
+		admin.declareExchange(exchange);
+		admin.declareQueue(queue);
+		// queue name and routing key are the same
+		admin.declareBinding(BindingBuilder.bind(queue).to(exchange).with(queueName));
+		final ConsumerSimpleMessageListenerContainer container = new ConsumerSimpleMessageListenerContainer();
+		container.setConsumerTagStrategy(q -> consumerTag);
+		container.setConnectionFactory(connectionFactory);
+		container.setQueueNames(queueName);
+
+		container.setErrorHandler(t -> {
+			if (isStopping)
+			{
+				return;
+			}
+			logException(t, null);
+			stop();
+		});
+		container.setMessageListener(this);
+		container.startConsumers();
+
+		log("Connected to AMQP Server. Waiting for messages!", // summary
+				null, // text
+				"queueName=" + queueName + ", consumerTag=" + consumerTag, // reference
+				null); // exception
+	}
+
+	/**
+	 * Import xml document
+	 *
+	 * @param xml xml document (as string) to be imported
+	 */
+	private void importXMLDocument(final String xml)
+	{
+		final Document documentToBeImported;
+
+		try
+		{
+			documentToBeImported = XMLHelper.createDocumentFromString(xml);
+		}
+		catch (final Exception e)
+		{
+			throw new AdempiereException(e);
+		}
+
+		final Mutable<Document> responseDOM = new Mutable<>(); // 03132
+
+		Services.get(ITrxManager.class).run(trxName, trxName -> {
+			final IImportHelper impHelper = Services.get(IIMPProcessorBL.class).createImportHelper(ctx);
+			try
+			{
+				final StringBuilder result = new StringBuilder();
+				responseDOM.setValue(impHelper.importXMLDocument(result, documentToBeImported, trxName));
+			}
+			catch (final Exception e)
+			{
+				throw e instanceof AdempiereException ? (AdempiereException)e : new AdempiereException(e);
+			}
+		});
+
+	}
+
+	public void stop()
+	{
+		isStopping = true;
+		try
+		{
+			log.trace("Closing AMQP Connection!");
+			if (connectionFactory != null)
+			{
+				connectionFactory.destroy();
+				connectionFactory.stop();
+				connectionFactory = null;
+			}
+		}
+		finally
+		{
+			isStopping = false;
+
+			// 02486 make sure that we inform 'replicationProcessor' so there is a change of being restarted
+			replicationProcessor.setProcessRunning(false);
+		}
+	}
+
+	private void logException(final Throwable e, final String messageText)
+	{
+		log("Error: " + e.getLocalizedMessage(),
```

Comment:
I think we can use ILoggable to the logging to both loggers and import processor log

Reply:
in order to use `ILoggable` afaiu you need to 
* create an `ILoggable` implementation that delegates to `Services.get(IIMPProcessorBL.class).createLog();`
* find a place where you can call `Loggables.temporarySetLoggable(ILoggable)` such that the BL that is running during import and export get's your ILogger

Reply:
@haneloreianoseck pls check out this commit: https://github.com/metasfresh/metasfresh/commit/7e57d975e44cb7a45f89933912610b66b627fb61
wdyt?

Reply:
Note:
> find a place where you can call Loggables.temporarySetLoggable(ILoggable) such that the BL that is running during import and export get's your ILogger

on export we don't log to and sortof audit table..right? so there afais using ILoggable doesn't make sense